addOns/postman/src/main/java/org/zaproxy/addon/postman/PostmanParser.java

```diff
@@ -75,15 +96,254 @@ public void importFromUrl(final String url) throws IllegalArgumentException, IOE
+            return httpMessage;
+        }
+
+        String mode = body.getMode();
+        if (mode == null) {
+            return httpMessage;
+        }
+
+        String bodyContent = "";
+        String contentType = "";
+
+        if (mode.equals(Body.RAW)) {
+            if (body.getRaw() == null) {
+                return httpMessage;
+            }
+
+            Map<String, String> contentTypeMap =
+                    Map.of(
+                            "html", "text/html",
+                            "javascript", "application/javascript",
+                            "json", "application/json",
+                            "xml", "application/xml");
+
+            contentType = "text/plain";
+
+            if (body.getOptions() != null && body.getOptions().getRaw() != null) {
+                String language = body.getOptions().getRaw().getLanguage();
+
+                if (language != null) {
+                    contentType =
+                            contentTypeMap.getOrDefault(
+                                    language.toLowerCase(Locale.ROOT), "text/html");
+                }
+            }
+
+            bodyContent = body.getRaw();
+        } else if (mode.equals(Body.URL_ENCODED)) {
+            if (body.getUrlencoded() == null) {
+                return httpMessage;
+            }
+
+            contentType = HttpHeader.FORM_URLENCODED_CONTENT_TYPE;
+
+            StringBuilder urlencodedBodySB = new StringBuilder();
+
+            for (KeyValueData data : body.getUrlencoded()) {
+                if (!data.isDisabled()) {
+                    if (urlencodedBodySB.length() > 0) {
+                        urlencodedBodySB.append('&');
+                    }
+                    try {
+                        urlencodedBodySB
+                                .append(
+                                        URLEncoder.encode(
+                                                data.getKey(), StandardCharsets.UTF_8.name()))
+                                .append('=')
+                                .append(
+                                        URLEncoder.encode(
+                                                data.getValue(), StandardCharsets.UTF_8.name()));
+                    } catch (UnsupportedEncodingException e) {
+                    }
+                }
+            }
+
+            bodyContent = urlencodedBodySB.toString();
+        } else if (mode.equals(Body.FORM_DATA)) {
+            if (body.getFormData() == null) {
+                return httpMessage;
+            }
+
+            String boundary = "----" + System.currentTimeMillis();
+
+            contentType = "multipart/form-data; boundary=" + boundary;
+
+            StringBuilder formDataBodySB = new StringBuilder();
+            for (FormData formData : body.getFormData()) {
+                if (!formData.isDisabled()) {
+                    formDataBodySB
+                            .append(generateMultiPartBody(formData, boundary))
```

Comment:
Would be better to pass the `StringBuilder`, though not needed for this PR (requires more changes).

Reply:
Just want to confirm, do you mean that I should have passed the `formDataBodySB` to `generateMultiPartBody` to let it handle the multipart body generation instead of having to use it for every `FormData` in a loop?

Reply:
Yes, to reuse it inside the method, instead of appending outside a new string it would append directly to it (less string allocations).